contracts/core/Core.sol

```diff
@@ -219,4 +234,87 @@ contract Core is
             _quantity
         );
     }
+
+    /**
+     * Function to convert Set Tokens into underlying components
+     *
+     * @param _tokenAddress    The address of the Set token
+     * @param _quantity        The number of tokens to redeem
+     */
+    function redeem(
+        address _tokenAddress,
+        uint _quantity
+    )
+        public
+        isValidTokenAddress(_tokenAddress)
+        isNonZero(_quantity)
+    {
+        // Check if multiple of naturalUnit
```

Comment:
We can make this check a private function, since it'll be used in redeem too

Reply:
isn't it already`internal`?